testnet/stacks-node/src/nakamoto_node/signer_coordinator.rs

```diff
@@ -336,10 +358,29 @@ impl SignerCoordinator {
                         return Err(NakamotoNodeError::BurnchainTipChanged);
                     }
 
+                    if rejections_timer.elapsed() > rejections_timeout {
+                        return Err(NakamotoNodeError::SigningCoordinatorFailure(
+                            "Gave up while tried reaching the threshold".into(),
+                        ));
+                    }
+
                     continue;
                 }
             };
 
+            if rejections != block_status.total_reject_weight {
+                rejections_timer = Instant::now();
+                rejections = block_status.total_reject_weight;
+                rejections_timeout = Duration::from_secs_f32(
+                    BLOCK_REJECTIONS_TIMEOUT_BASE as f32
+                        - (BLOCK_REJECTIONS_TIMEOUT_BASE as f32
+                            * ((rejections as f32 / self.weight_threshold as f32).powf(2.0))),
+                );
```

Comment:
So, I think a better strategy might be to just define a stepping function.

Like,

```rust
let reject_threshold = self.total_weight.saturating_sub(self.weight_threshold);
let block_rejection_timeout_steps = BTreeMap::from([
   (0, 600),
   (reject_threshold / 3, 300),
   (reject_threshold * 2 / 3, 150),
   (reject_threshold, 0)
]);

let rejection_timeout_secs = block_rejection_timeout_steps
     .range((Included(0), Included(rejections)))
     .last()
     .ok_or_else(|| NakamotoNodeError::SignerCoordinatorFailure("Invalid rejection timeout step function definition".into()))?;
```

Then, I think we just make that all configurable via the node configuration -- that way we don't need a test static.

Reply:
So, you would like to have 3 config parameters for the signer for each of the thresholds ? or you would prefer a single array-like option where you define each step ? (so you can have more than 3, or 1 eventually).

Reply:
Yeah -- I think a single array-like option.